src/api/app/controllers/request_controller.rb

```diff
@@ -161,6 +165,10 @@ def request_create
 
   def request_command_diff
     req = BsRequest.find_by_number!(params[:id])
+    superseded_request = req.superseding.find_by_number(params[:diff_to_superseded])
+    if params[:diff_to_superseded].present? && superseded_request.blank?
+      raise SupersededRequestNotFound
```

Comment:
:thinking: You could directly call [render_error](https://github.com/openSUSE/open-build-service/blob/master/src/api/app/controllers/application_controller.rb#L322) here.